File: src/main/kotlin/ympa_kotlin_wiremock_server/models/GoodsStatsWarehouseDTO.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

data class GoodsStatsWarehouseDTO(
    @field:JsonProperty("stocks")
    val stocks: kotlin.collections.List<WarehouseStockDTO>,

    @field:JsonProperty("id")
    val id: kotlin.Long? = null,

    @field:JsonProperty("name")
    val name: kotlin.String? = null,

)

File: src/App/DTO/GoodsStatsWarehouseDTO.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * Информация о складе.
 */
class GoodsStatsWarehouseDTO
{
    /**
     * Идентификатор склада.
     * @DTA\Data(field="id", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     * @var int|null
     */
    public $id;

    /**
     * Название склада.
     * @DTA\Data(field="name", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     * @var string|null
     */
    public $name;

    /**
     * Информация об остатках товаров на складе.
     * @DTA\Data(field="stocks")
     * @DTA\Strategy(name="Object", options={"type":::class})
     * @DTA\Validator(name="TypeCompliant", options={"type":::class})
     * @var \App\DTO\WarehouseStockDTO[]|null
     */
    public $stocks;

}
